servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/LoadBalancerObserver.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright © 2023 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.loadbalancer;
+
+import io.servicetalk.client.api.NoActiveHostException;
+import io.servicetalk.client.api.ServiceDiscovererEvent;
+
+import java.util.Collection;
+import javax.annotation.Nullable;
+
+/**
+ * An observer that provides visibility into a {@link io.servicetalk.client.api.LoadBalancer}
+ * @param <ResolvedAddress> the type of the resolved address.
+ */
+interface LoadBalancerObserver<ResolvedAddress> {
```

Comment:
public

Reply:
I'd prefer to keep it private until we are a bit more confident, although that is kind of a chicken and egg problem. Can we tag it with the `@UnstableApi` annotation or something or is that a no-go?

Reply:
The typical way so-far was the `experimental` package. We can have such a package and move it out when we are ready.